File: language-support/js/daml-ledger/src/validation/get_transaction_by_id_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { array, native, object, Validation } from "./base";
import { RequiredValidation } from "./base/typelevel";

function required(): RequiredValidation<ledger.GetTransactionByIdRequest> {
    return {
        transactionId: native('string'),
        requestingParties: array(native('string')),
    };
}

export const GetTransactionByIdRequest: Validation = object<ledger.GetTransactionByIdRequest>('GetTransactionByIdRequest', required, () => ({}));

File: ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/messages/transaction/GetTransactionByIdRequest.scala
// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.api.messages.transaction

import com.daml.ledger.api.domain.{LedgerId, TransactionId}
import com.daml.lf.data.Ref.Party

final case class GetTransactionByIdRequest(
    ledgerId: Option[LedgerId],
    transactionId: TransactionId,
    requestingParties: Set[Party],
)
